src/extensions/Statiq.ApiClient/ReadApiWithClient.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Statiq.Common;
+
+namespace Statiq.ApiClient
+{
+    /// <summary>
+    /// Outputs metadata for information from any API with its client.
+    /// </summary>
+    /// <remarks>
+    /// This modules used to submit requests to the API with the API client. Because of the possible large
+    /// number of different kinds of requests, this module does not attempt to provide a fully abstract wrapper
+    /// around the API. Instead, it simplifies the housekeeping involved in setting up an
+    /// API client and requires you to provide functions that fetch whatever data you need. Each request
+    /// will be sent for each input document.
+    /// </remarks>
+    /// <category>Metadata</category>
+    public class ReadApiWithClient<TClient> : ParallelSyncModule
```

Comment:
> I think `ReadApi<TClient>` might be a better name - and fits with some of the other Read... modules. I know it doesn't _exactly_ describe what this module does, but I think it's close enough.

I originally chose this name, but since it doesn't reveal the essence of the module, I decided to change it. I agree, should rename it.

Reply:
> Should this module be a base module for the other new ones like `ReadTelegram`? If so we could consider combining them all into a single PR, or I could get this one merged and you could update those ones with this module as the base.

It is a good idea! I can combine all of modules into one new PR with `ReadApi<TClient>` as base module after you merge it.